src/System.Linq/tests/SingleTests.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System.Linq.Tests.Helpers;
+using Xunit;
+
+namespace System.Linq.Tests
+{
+    public class SingleTests
+    {
+        [Fact]
+        public void FailOnEmpty()
+        {
+            Assert.Throws<InvalidOperationException>(() => Enumerable.Empty<int>().Single());
+        }
+
+        [Fact]
+        public void DefaultOnEmpty()
+        {
+            Assert.Equal(0, Enumerable.Empty<int>().SingleOrDefault());
+        }
+
+        [Fact(Skip="Runs for thousands of years. Then throws wrong exception")]
+        public void FailOnManyMatches()
+        {
+            Assert.Throws<InvalidOperationException>(() => Enumerable.Range(0, int.MaxValue).Single(i => i > 0));
```

Comment:
I think I mentioned this in the previous PR, but I'd prefer to see this done differently, testing the current behavior and not being disabled, e.g.
```
[Fact]
public void FailOnManyMatches()
{
    int invocations = 0;
    Assert.Throws<InvalidOperationException>(() => Enumerable.Range(1, 10).Single(i => i > 0));
    Assert.Equal(10, invocations);
}
```
If we decide to take a visible behavioral change that introduces short circuiting, then the test can be updated at the same time, but at least this will help to flag such behavioral changes when they're attempted.